tools/visual-pipeline-and-platform-evaluation-tool/vippet/utils.py

```diff
@@ -1,50 +1,85 @@
+) -> str:
     """
-    Checks if the given model path corresponds to a YOLO v10 model.
+    Replace all tee names in the pipeline string with unique names based on pipeline and stream indices.
+
+    GStreamer pipelines may contain multiple tees with different names (e.g., tee name=t0, tee name=t1).
+    Each tee is referenced later in the pipeline (e.g., t0., t1.).
+    This function ensures each tee gets a unique name to avoid conflicts when combining multiple pipelines.
 
     Args:
-        model_path (str): Path to the model file.
+        pipeline_str: The GStreamer pipeline string containing tee elements.
+        pipeline_index: The index of the pipeline in the run specification list.
+        stream_index: The index of the stream within the pipeline.
 
     Returns:
-        bool: True if the model is a YOLO v10 model, False otherwise.
+        str: The pipeline string with all tee names replaced with unique identifiers.
+
+    Example:
+        Input:  "... tee name=t0 ! queue t0. ! ..."
+        Output: "... tee name=t100 ! queue t100. ! ..."
     """
-    return "yolov10" in model_path.lower()
+    logger.debug(
+        f"Making tee names unique for pipeline_index={pipeline_index}, stream_index={stream_index}"
+    )
 
+    # Find all unique tee names in the pipeline using regex
+    # Pattern matches "tee name=<name>" where <name> is typically t0, t1, t2, etc.
+    tee_pattern = r"tee\s+name=(\w+)"
+    tee_matches = re.finditer(tee_pattern, pipeline_str)
 
-def download_file(url, local_filename):
-    file_path = os.path.join(TEMP_DIR, local_filename)
-    if os.path.exists(file_path):
-        logger.info(f"File {file_path} already exists, skipping download.")
-        return file_path
+    # Collect all unique tee names
+    tee_names = set()
+    for match in tee_matches:
+        tee_names.add(match.group(1))
 
-    # Send a GET request to the URL
-    with requests.get(url, stream=True) as response:
-        response.raise_for_status()  # Check if the request was successful
-        # Open a local file with write-binary mode
-        with open(os.path.join(TEMP_DIR, local_filename), "wb") as file:
-            # Iterate over the response content in chunks
-            for chunk in response.iter_content(chunk_size=8192):
-                file.write(chunk)  # Write each chunk to the local file
-    return file_path
+    if tee_names:
+        logger.debug(f"Found {len(tee_names)} unique tee(s): {sorted(tee_names)}")
+    else:
+        logger.debug("No tee elements found in pipeline")
+        return pipeline_str
 
+    # Replace each tee name with a unique identifier
+    for original_name in tee_names:
+        # Extract digits from original name (e.g., "t0" -> "0", "t1" -> "1")
+        original_digits = "".join(filter(str.isdigit, original_name))
+        # Create unique name: t{pipeline_index}{stream_index}{original_digits}
+        new_name = f"t{pipeline_index}{stream_index}{original_digits}"
```

Comment:
The current `make_tee_names_unique` implementation derives the new tee name only from the digits present in the original name. If we have multiple tees whose names do not contain digits (e.g. `tee name=a`, `tee name=b`) they can collapse to the same generated name and conflict.
To avoid potential collisions, it is safer to include a local index when generating the new name, so each tee in a pipeline is guaranteed to get a unique suffix even if their original names share the same digit pattern (or have no digits at all).

```suggestion
    # Replace each tee name with a unique identifier.
    # Using an index in addition to digits from the original name prevents
    # collisions when multiple tees have names without digits or share
    # the same digit pattern.
    for idx, original_name in enumerate(sorted(tee_names)):
        # Extract digits from original name (e.g., "t0" -> "0", "t1" -> "1")
        original_digits = "".join(filter(str.isdigit, original_name))
        # Create unique name: t{pipeline_index}{stream_index}{idx}{original_digits}
        new_name = f"t{pipeline_index}{stream_index}{idx}{original_digits}"
```